builtin/providers/aws/provider.go

```diff
@@ -32,6 +32,15 @@ func Provider() terraform.ResourceProvider {
 				Description: descriptions["secret_key"],
 			},
 
+			"token": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{
+					"AWS_SESSION_TOKEN",
+				}, nil),
```

Comment:
Set this `nil` to `""` to prevent the interactive input UI from asking for it. Its unlikely if you're using interactive input that you'll have this.